app/tests/profiles_tests/test_views.py

```diff
@@ -179,3 +180,74 @@ def test_organization_update(self, client):
         )
 
         assert org1.title not in response.content.decode()
+
+
+@pytest.mark.parametrize(
+    "viewname,subscription_attr",
+    (
+        ("newsletter-unsubscribe", "receive_newsletter"),
+        ("notification-unsubscribe", "receive_notification_emails"),
+    ),
+)
+@pytest.mark.django_db
+def test_one_click_unsubscribe(client, settings, viewname, subscription_attr):
+    # Override the celery settings
+    settings.task_eager_propagates = (True,)
+    settings.task_always_eager = (True,)
+
+    user = UserFactory()
+    user.user_profile.receive_newsletter = True
+    user.user_profile.receive_notification_emails = True
+    user.user_profile.save()
+
+    token = user.user_profile.unsubscribe_token
+
+    # User does not need to be logged in to unsubscribe
+    response = get_view_for_user(
+        client=client,
+        viewname=viewname,
+        reverse_kwargs={"token": token},
+        method=client.post,
+    )
+    assert response.status_code == 302
+    user.user_profile.refresh_from_db()
+    assert not getattr(user.user_profile, subscription_attr)
+
+    # reset preferences
```

Comment:
Try not to create long tests unless necessary. Debugging very long tests can hide errors. You should try for the setup - modify - assert pattern. Some people go to extremes and only have one assert per test but I think there is a halfway house. That you have 3 comments in here, and that you have a reset, suggests that there should be 3 separate tests. Have a look at some of the things Adam Johnson has put out (e.g. https://adamj.eu/tech/2020/06/15/how-to-unit-test-a-django-form/).